src/OrganisationUnitHierarchy/OrganisationUnitHierarchy.component.js

```diff
@@ -179,42 +180,38 @@ function moveOrganisationUnit() {
         .subscribe(
             (message) => {
                 snackActions.show({ message, translate: false });
-
-                const hierarchy = Object.assign({}, appState.state.hierarchy);
+            },
+            (e) => {
+                setHierarchyProcessingStatus(appState.state.hierarchy, false);
+            },
+            () => {
                 setAppState({
                     hierarchy: {
-                        ...hierarchy,
-                        isProcessing: false,
+                        ...appState.state.hierarchy,
+                        selectedLeft: [],
+                        selectedRight: [],
+                        initiallySelected: [],
+                        moveTargetPath: appState.state.hierarchy.selectedRight[0].path,
                         reload: []
-                            .concat(hierarchy.selectedRight.map(model => model.id))
-                            .concat(hierarchy.selectedLeft.map((model) => {
+                            .concat(appState.state.hierarchy.selectedRight.map(model => model.id))
+                            .concat(appState.state.hierarchy.selectedLeft.map((model) => {
                                 if (model.parent && model.parent.id) {
                                     return model.parent.id;
                                 }
                                 return appState.state.hierarchy.leftRoots
                                     .concat(appState.state.hierarchy.rightRoots)
                                     .map(value => value.id)
                                     .filter(rootId => new RegExp(rootId).test(model.path))
-                                    .reduce(value => value);
+                                    .reduce(value => value)
                             })),
                     },
                 });
-            },
-            (e) => {
-                setHierarchyProcessingStatus(appState.state.hierarchy, false);
-            },
-            () => {
-                const hierarchy = Object.assign({}, appState.state.hierarchy);
 
-                hierarchy.selectedLeft = [];
-                hierarchy.selectedRight = [];
-                hierarchy.initiallySelected = [];
-                setAppState({
-                    hierarchy: {
-                        ...hierarchy,
-                        isProcessing: false,
-                    },
-                });
+                // Wait till stack clears before setting isProcessing to false and triggering a rerender
+                // Without this the UI represents a stale state
+                setTimeout(() => {
+                    setHierarchyProcessingStatus(appState.state.hierarchy, false)
+                }, 0);
```

Comment:
I've tried to reduce the number of calls to `setAppState`, but for reasons unknown to me, the UI is not updating properly without this defer....